geth-utils/gethutil/trace.go

```diff
@@ -215,9 +217,9 @@ func Trace(config TraceConfig) ([]*ExecutionResult, error) {
 	executionResults := make([]*ExecutionResult, len(config.Transactions))
 	for i, message := range messages {
 		tracer := logger.NewStructLogger(config.LoggerConfig)
-		evm := vm.NewEVM(blockCtx, core.NewEVMTxContext(message), stateDB, &chainConfig, vm.Config{Debug: true, Tracer: tracer, NoBaseFee: true})
+		evm := vm.NewEVM(blockCtx, core.NewEVMTxContext(&message), stateDB, &chainConfig, vm.Config{Debug: true, Tracer: tracer, NoBaseFee: true})
 
-		result, err := core.ApplyMessage(evm, message, new(core.GasPool).AddGas(message.Gas()))
+		result, err := core.ApplyMessage(evm, &message, new(core.GasPool).AddGas(message.GasLimit))
```

Comment:
I think these changes are correct based on the upstream diff: https://github.com/ethereum/go-ethereum/commit/67ac5f0ae797fbafa1e57e982cab4c2d291244c3